src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt

```diff
@@ -232,6 +231,32 @@ class UserService(
     userPersonDetailRepository.findById(username).orElseThrow(UserNotFoundException("User $username not found"))
     userPersonDetailRepository.changePassword(username, password)
   }
+
+  fun setDefaultCaseload(username: String, defaultCaseloadId: String): UserCaseloadDetail {
+    val user = userPersonDetailRepository.findById(username).orElseThrow(UserNotFoundException("User $username not found"))
+    user.setDefaultCaseload(defaultCaseloadId)
+    return user.toUserCaseloadDetail()
+  }
+
+  fun addCaseloadToUser(username: String, caseloadId: String): UserCaseloadDetail {
+    val user = userPersonDetailRepository.findById(username).orElseThrow(UserNotFoundException("User $username not found"))
+
+    val caseload = caseloadRepository.findById(caseloadId)
+      .orElseThrow(CaseloadNotFoundException("Caseload $caseloadId not found"))
+
+    user.addCaseload(caseload)
+    return user.toUserCaseloadDetail()
+  }
+
+  fun getCaseloads(username: String): UserCaseloadDetail {
+    return userPersonDetailRepository.findById(username).orElseThrow(UserNotFoundException("User $username not found")).toUserCaseloadDetail()
```

Comment:
is there any advantage on adding `@transactional(readonly = true)` to the read only calls? We do in some places but not sure what affect it actually has